Modules/_asynciomodule.c

```diff
@@ -193,6 +193,44 @@ is_coroutine(PyObject *coro)
     return has_it;
 }
 
+/*[clinic input]
+_asyncio.isfuture
+
+    obj: object
+
+[clinic start generated code]*/
+
+static PyObject *
+_asyncio_isfuture_impl(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=7ceb4ecaca613a86 input=3c9f9afc85c35bc5]*/
+{
+  _Py_IDENTIFIER(__class__);
+  PyObject* class = _PyObject_GetAttrId(obj, &PyId___class__);
+  if (class == NULL) {
+    return NULL;
```

Comment:
Please use 4 spaces to ident your code (as the rest of `_asynciomodule.c` file).